src/monodroid/jni/android-system.cc

```diff
@@ -266,7 +265,7 @@ AndroidSystem::monodroid_read_file_into_memory (const char *path, char **value)
 			r = ADD_WITH_OVERFLOW_CHECK (size_t, static_cast<size_t>(fileStat.st_size), 1);
 			if (value && (*value = new char[r])) {
 				size_t nread = fread (*value, 1, static_cast<size_t>(fileStat.st_size), fp);
-				if (nread == 0 || nread != r) {
+				if (nread == 0 || nread != r - 1) {
```

Comment:
Instead of using `r-1`, which could slowly "drift" over time if any additional lines are added between the `r=ADD_WITH_OVERFLOW_CHECK()` and here, why not instead do `nread != fileStat.st_size`?